M2/Macaulay2/packages/Varieties.m2

```diff
@@ -151,11 +205,10 @@ isProjective ProjectiveVariety := X -> true
 -- TODO: instead of an error, return Proj R when there is no variety,
 -- then replace Proj ring M in code for sheaf with variety ring M
 variety = method(TypicalValue => Variety)
-variety Ring        := S -> if S.?variety then S.variety else error "no variety associated with ring"
-variety Ideal       := I -> Proj(ring I/I) -- TODO: should this be saturated?
-variety RingElement := f -> variety ring f -- TODO: should this be V(f) instead?
+variety Ring  := S -> if S.?variety then S.variety else error "no variety associated with ring"
+variety Ideal := I -> Proj quotient I -- TODO: should this be saturated?
```

Comment:
I think that common interpretation for the variety of an ideal depends significantly on whether it represents something in projective space or affine space.  Just looking at the ideal seems to be insufficient information.  Specifically, I would be tempted to define ```subvariety(Ideal, Variety)``` rather than simply ```variety Ideal```.

Reply:
I'll add a comment to this effect, but I won't change this now because this was simply ported over from old `varieties.m2` and I think we should discuss it more before introducing a potentially breaking change.